src/DevServer/index.ts

```diff
@@ -176,10 +180,17 @@ export class DevServer {
      */
     this.logger.info('building project...')
 
+    /**
+     * Parse some environment variables
+     */
+    const env = new EnvParser()
+      .parse(this.application.appRoot)
+      .asEnvObject(['PORT', 'TZ', 'HOST', 'ASSETS_DRIVER'])
```

Comment:
I'm not sure this is the best way to get the assets driver. For me, it should be retrieved from the config. But for that we need to load the application and the IoC. What I tried to do, but I got a strange error.

I guess it's because the assembler command manifest is loaded in a special way? Let me know if there is a way

Reply:
Nope, using the environment variable is fine. We can make a hard requirement to toggle the assets driver with env var only and there is no config.

**Now the question might be, isn't it bad to rely on Environment directly?**
But, we and Node itself relies on many already. `TZ` is used by Node, `NODE_ENV`, `PORT` is used by the entire ecosystem.

Reply:
Yes, I am not sure using an environment variable to configure this feature would be the best option.

Those are used when you need different values depending on your environment. I would go with something directly in the configuration file.

Reply:
What if we detect it? If there is a `webpack.config.js` file, then we will use encore and the there is `vite.config.js` file, then we use vite.

Finally, you can define it within the `.adonisrc.json` file, if automatic detection is not what you want?

Reply:
I would say that I prefer to keep the environment variable for the sake of consistency

Setting it in the rc file, seems weird
And for the detection, if both config files are present what do we do? Which one takes priority? Do we fallback on the environment variable? 

But just to be sure, do we agree that it is not possible to access the configuration from this place? Because we already have a config object for the assets manager in `config/app.ts`

Reply:
If they have both and the configuration is in `auto`, we can output an error.

I do not see any gains by using an environment variable, this is something you will probably never change, and it is not linked to the environment.

Also, I am not seeing where we lose consistency by not having an environment variable for this feature.

Reply:
> I do not see any gains by using an environment variable, this is something you will probably never change, and it is not linked to the environment.
Agree

But I say that we lose consistency because some drivers are also defined in the environment file. Then they are used in the config, but well, they are first in the .env file

Reply:
> some drivers are also defined in the environment file

Yes, but those are linked to the application runtime and can change depending on the environment.
That is not the case with the bundler.

Reply:
Yeah, you're right. Well, so lets's go for the RC file if that's okay with you guys!

I'll add this to the JSON schema of the vscode extension

Reply:
Yeah, I accept the argument of @RomainLanz. Also, in more than 80% of the cases, we will be able to just detect it.

It will be nice to put a log message telling how we picked the assets driver. Something like.

```
Found "vite.config.js" file. Using Vite as the assets driver
```

Or 

```
Assets driver set to "vite" in ".adonisrc.json"
```

Reply:
Was added exactly this way. Thanks guys !